etl/steps/data/meadow/hmd/2022-11-03/hmd_lt.py

```diff
@@ -0,0 +1,322 @@
+        # Check available files
+        _sanity_check_files(tmp_dir)
+
+        # Load data
+        for i in [1, 5]:
+            for j in [1, 5, 10]:
+                # Create table and folder name
+                table_name = f"{i}x{j}"
+                # Create table
+                log.info(f"Creating table '{table_name}'...")
+                table = make_table(tmp_dir, table_name, walden_ds)
+                # add table to a dataset
+                log.info("Adding table to dataset...")
+                ds.add(table)
+    return ds
+
+
+def _sanity_check_files(path: str) -> None:
+    """Checks that all required files are present once zip is uncompressed."""
+    files_found = sorted(os.listdir(path))
+    assert (
+        files_found == FILES_EXPECTED
+    ), f"Files found are not the ones expected! Check that {FILES_EXPECTED} are actually there!"
+
+
+def make_table(input_folder: str, table_name: str, walden_ds: WaldenCatalog) -> catalog.Table:
+    """Create table.
+
+    Loads data from `input_folder` and creates a table with the name `table_name`. It uses the metadata from `walden_ds`.
+
+    Parameters
+    ----------
+    walden_ds : WaldenCatalog
+        Walden dataset.
+    input_folder : str
+        Folder containing uncompressed data from Walden.
+    table_name : str
+        Name of the table.
+
+    Returns
+    -------
+    catalog.Table
+        Table with data.
+    """
+    # Load files
+    f = f"bltper_{table_name}"
+    files = glob(os.path.join(input_folder, f"{f}/*.txt"))
+    log.info(f"Looking for available files in {f}...")
+    # Create df
+    df = make_df(files, table_name)
+    # Clean df
+    df = clean_df(df)
+    # df to table
+    table = df_to_table(walden_ds, table_name, df)
+    # underscore all table columns
+    table = underscore_table(table)
+    return table
+
+
+def make_df(files: List[str], table_name: str) -> pd.DataFrame:
+    """Create dataframe.
+
+    Parameters
+    ----------
+    files : List[str]
+        Files to load and extract data from. There is a file per country. Within the file, data is found
+        along with country name. Note that sometimes, the country name can be like "New Zealand - Non-Maori".
+    table_name : str
+        Name of the table.
+
+    Returns
+    -------
+    pd.DataFrame
+        _description_
+    """
+    log.info("Creating dataframe from files...")
+    regex = FILE_REGEX.format(table=table_name)
+    dfs = []
+    # Load each file
+    for f in files:
+        with open(f, "r") as f:
+            text = f.read()
+        # Get relevant fields
+        match = re.search(regex, text)
+        if match is not None:
+            country, _, table_str = match.group(1, 2, 3)
+        else:
+            raise ValueError("No match found!")
+        # Build country df
+        df_ = _make_df_country(country, table_str)
+        dfs.append(df_)
+    # Concatenate all country dfs
+    df = pd.concat(dfs, ignore_index=True)
+    return df  # type: ignore
```

Comment:
Instead of ignoring type checks, you can do `return cast(pd.DataFrame, df)` (and add `from typing import cast`).